test/jdk/tools/jpackage/macosx/CustomInfoPListTest.java

```diff
@@ -185,7 +186,8 @@ ThrowingConsumer<JPackageCommand> createPListFilesVerifier(JPackageCommand cmd)
             if (defaultPListFiles.isEmpty()) {
                 return defaultVerifier;
             } else {
-                var vanillaCmd = new JPackageCommand().setFakeRuntime()
+                boolean includeBin = customPLists.contains(CustomPListType.EMBEDDED_RUNTIME_WITH_BIN);
```

Comment:
`customPLists.contains(CustomPListType.EMBEDDED_RUNTIME_WITH_BIN)` puts restrictions on what test cases can be executed with the external runtime that has the "bin" subdirectory.

This should be a separate property of TestConfig class:
```
public record TestConfig(Set<CustomPListType> customPLists, boolean runtimeWithBinDir)
```